db/migrate/20170609214859_optional_required_notes_field.rb

```diff
@@ -0,0 +1,27 @@
+class OptionalRequiredNotesField < ActiveRecord::Migration
+
+  class Product < ActiveRecord::Base
+  end
+
+  def up
+    add_column :products, :user_notes_field_mode, :string, null: false, default: "hidden"
+
+    Product.find_each do |product|
+      mode = product.note_available_to_users? ? "optional" : "hidden"
+      product.update_column(:user_notes_field_mode, mode)
+    end
+
+    remove_column :products, :note_available_to_users
```

Comment:
Separate migrations for the drop column might be overkill in this case, but do we want this in a transaction block? So it'll rollback/not drop the column if the data migration somehow fails?